File: python/programmers/stack_queue/주식가격.py
def solution(prices):
    answer = []
    for i in range(len(prices)):
        sec = 0
        for j in range(i, len(prices)):
            if i != j:
                sec += 1
            if prices[i] > prices[j]:
                break
        answer.append(sec)
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/주식가격.java
package ps_traning.programmers.level_2;

import java.util.Stack;

public class 주식가격 {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < prices.length; i++) {
            while (!stack.isEmpty() && prices[stack.peek()] > prices[i]) {
                int index = stack.pop();
                answer[index] = i - index;
            }
            stack.push(i);
        }
        while (!stack.isEmpty()) {
            int index = stack.pop();
            answer[index] = prices.length - 1 - index;
        }

        return answer;
    }
}
